### PR TITLE
EPMRPP-77637 || Notifications actions update

### PR DESCRIPTION
**Pull request by tr1ble**

### CODE REVIEW

File: app/localization/translated/uk.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
   "AddEditNotificationCaseModal.launchOwnerLabel": "Запустити власника (хто запустив - той отримує)",
   "AddEditNotificationCaseModal.launchesHint": "Ім’я запуску повинно бути довжиною від 3 до 256.",
   "AddEditNotificationCaseModal.newRuleMessage": "Додати",
+  "AddEditNotificationCaseModal.copyRuleMessage": "Дублікат",
```
Comment (chivekrodis) [inline app/localization/translated/uk.json:66]: It's a noun but should be verb

File: app/src/common/urls.js [hunk 2]
```diff
@@ -181,6 +181,8 @@ export const URLS = {
   suite: (activeProject, suiteId) => `${urlBase}${activeProject}/item/${suiteId}`,
 
   notification: (activeProject) => `${urlBase}notification/${activeProject}`,
+  deleteNotification: (activeProject, notificationId) =>
+    `${urlBase}notification/${activeProject}/${notificationId}`,
```
Comment (AmsterGet) [inline app/src/common/urls.js:185]: You don't use any 'delete' word in the url itself, so it must be self-explained, f.e.:
`notificationById`, as this endpoint can also be used to update notification.

File: app/src/controllers/project/actionCreators.js [hunk 3]
```diff
@@ -64,6 +66,16 @@ export const updateProjectNotificationsConfigAction = (config) => ({
...
+export const updateProjectNotificationAction = (notification) => ({
+  type: UPDATE_NOTIFICATION,
+  payload: notification,
+});
+
+export const deleteProjectNotificationAction = (id) => ({
+  type: DELETE_NOTIFICATION,
+  payload: id,
+});
+
```
Comment (AmsterGet) [inline app/src/controllers/project/actionCreators.js:78]: Please move this actions down to the other new actions added in the previous tasks.
And use action name as `UPDATE_PROJECT_NOTIFICATION`, as we have system notifications that are just `*_NOTIFICATION`

File: app/src/controllers/project/constants.js [hunk 4]
```diff
@@ -27,6 +27,8 @@ export const FETCH_PROJECT_PREFERENCES_SUCCESS = 'fetchProjectPreferencesSuccess
 export const FETCH_CONFIGURATION_ATTRIBUTES = 'fetchConfigurationAttributes';
 export const UPDATE_CONFIGURATION_ATTRIBUTES = 'updateConfigurationAttributes';
 export const UPDATE_NOTIFICATIONS_CONFIG = 'updateNotificationsConfig';
+export const UPDATE_NOTIFICATION = 'updateNotification';
+export const DELETE_NOTIFICATION = 'deleteNotification';
```
Comment (AmsterGet) [inline app/src/controllers/project/constants.js:31]: Move down.

File: app/src/controllers/project/sagas.js [hunk 5]
```diff
@@ -198,6 +200,75 @@ function* watchAddProjectNotification() {
...
+    const projectId = yield select(projectIdSelector);
+    const newConfig = {
+      ...currentConfig,
+      cases: currentConfig.cases.map((item) => {
+        if (notification.id === item.id) return notification;
+        return item;
+      }),
+    };
+
+    yield call(fetch, URLS.notification(projectId), {
```
Comment (AmsterGet) [inline app/src/controllers/project/sagas.js:216]: Shouldn't it be updated by `id`?

File: app/src/controllers/project/sagas.js [hunk 6]
```diff
@@ -198,6 +200,75 @@ function* watchAddProjectNotification() {
...
+  try {
+    const currentConfig = yield select(projectNotificationsConfigurationSelector);
+    const projectId = yield select(projectIdSelector);
+    const newConfig = {
+      ...currentConfig,
+      cases: currentConfig.cases.map((item) => {
+        if (notification.id === item.id) return notification;
+        return item;
+      }),
+    };
```
Comment (AmsterGet) [inline app/src/controllers/project/sagas.js:214]: Don't use legacy `cases`.

File: app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx [hunk 7]
```diff
@@ -137,6 +131,26 @@ export const Notifications = ({ setHeaderTitleNode }) => {
     );
   };
 
+  const onCopy = (notification) => {
+    trackEvent(SETTINGS_PAGE_EVENTS.CLONE_PATTERN_ICON);
```
Comment (AmsterGet) [inline app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx:135]: It is not pattern. It is a notification. Clarify with Web Analyst the corresponding event.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

**Review by chivekrodis — CHANGES_REQUESTED** [review]
Also what about attributes?
We have a flow to save them and right now if you fill attribute, uncheck attribute control and click save attributes appears in result

File: app/src/common/urls.js (hunk 2 above)
Comment (tr1ble) [reply app/src/common/urls.js:185]: Updating passes without ID,

File: app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx [hunk 8]
```diff
@@ -137,6 +131,26 @@ export const Notifications = ({ setHeaderTitleNode }) => {
...
+    const newNotification = {
+      ...notification,
+      ruleName: notification.ruleName + COPY_POSTFIX,
+    };
+    delete newNotification.id;
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx:140]: Or use `omit` util function.

File: app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx (hunk 7 above)
Comment (tr1ble) [reply app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx:135]: It's my typo, sorry

**Comment by tr1ble** [comment]
> Also what about attributes? We have a flow to save them and right now if you fill attribute, uncheck attribute control and click save attributes appears in result

I would fix it with new component

File: app/src/controllers/project/actionCreators.js [hunk 9]
```diff
@@ -176,6 +170,15 @@ export const updatePAStateAction = (PAState) => ({
   payload: PAState,
 });
 
+export const fetchNotificationsPageAction = () => ({
```
Comment (AmsterGet) [inline app/src/controllers/project/actionCreators.js:173]: Better `fetchProjectNotificationsAction`

File: app/src/controllers/project/actionCreators.js [hunk 10]
```diff
@@ -176,6 +170,15 @@ export const updatePAStateAction = (PAState) => ({
...
+  type: FETCH_NOTIFICATIONS_PAGE,
+});
+
+export const fetchNotificationsPageSuccessAction = (notifications) => ({
+  type: FETCH_PROJECT_NOTIFICATIONS_SUCCESS,
```
Comment (AmsterGet) [inline app/src/controllers/project/actionCreators.js:178]: Please keep action type and action creator names consistent.

File: app/src/controllers/project/reducer.js [hunk 11]
```diff
@@ -169,8 +161,37 @@ export const projectInfoLoadingReducer = (state = false, { type }) => {
   }
 };
 
+export const projectNotificationsReducer = (state = {}, { type, payload }) => {
+  console.log(type);
```
Comment (AmsterGet) [inline app/src/controllers/project/reducer.js:165]: Recheck the commit before pushing it.

File: app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx [hunk 12]
```diff
@@ -132,20 +139,20 @@ export const Notifications = ({ setHeaderTitleNode }) => {
...
 
   const onCopy = (notification) => {
-    trackEvent(SETTINGS_PAGE_EVENTS.CLONE_PATTERN_ICON);
-    const newNotification = {
-      ...notification,
-      ruleName: notification.ruleName + COPY_POSTFIX,
-    };
+    trackEvent(SETTINGS_PAGE_EVENTS.CLONE_NOTIFICATIONS);
+    const { id, ...newNotification } = notification;
     delete newNotification.id;
```
Comment (AmsterGet) [inline app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx:144]: Do you still need `delete newNotification.id;`?

**Review by AmsterGet — CHANGES_REQUESTED** [review]

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**